### PR TITLE
Extract shared artifacts from Alerting

### PR DESCRIPTION
**Pull request by lucasponce**
Common logging & properties utils
Rest Api for Vert.X/Netty handlers
Common Vert.X/Netty server loader

As commented @stefannegrea. 
Please could you review and merge ?
I have a local alerting branch working with them, I would need a release in orther to link the dependencies (or just a merge with temporal srcdeps).

/cc @jshaughn

### CODE REVIEW

**Comment by lucasponce** [comment]
In this local WIP branch
https://github.com/lucasponce/hawkular-alerts/tree/next
you can see how itests are working using the shared artifacts.

**Comment by jshaughn** [comment]
@lucasponce This all looks good to me, I think.  The short module naming is different than what is there already but I'm not opposed to it.  I like moving towards a consolidated properties file.  The term HawkularServer could maybe get confusing with Hawkular Services but it makes sense and I don't have a better name.  I think we should maybe hold on a release but I'm against a merge and the use of a src dep.  The other alternative would be be to create a "next" branch for commons as well, but that may be unnecessary.  I guess the main question is whether there is buy-in for vert.x stuff in commons.

**Comment by lucasponce** [comment]
I am afraid that for travis uses and several repos we need to rely on srcdeps or releases.
@stefannegrea commented that it would be good to use commons for shared artifacts, I think we can move further and refine in iterative steps.

**Comment by jshaughn** [comment]
LGTM, @stefannegrea, any objections to merge?

**Comment by stefannegrea** [comment]
@lucasponce, all these new modules should be placed under a folder since they are specific to a new rest implementation.

File: server/src/main/java/org/hawkular/HandlersManager.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                                RestEndpoint endpoint = (RestEndpoint)clazz.getAnnotation(RestEndpoint.class);
+                                for (int j=0; j<interfaces.length; j++) {
+                                    if (interfaces[j].equals(RestHandler.class)) {
+                                        endpointsClasses.put(endpoint.path(), clazz);
+                                    }
+                                }
+                            }
+                        } catch (Exception e) {
+                            log.errorf(e, "Error loading Handler [%s].", className);
+                            System.exit(1);
```
Comment (lucasponce) [inline server/src/main/java/org/hawkular/HandlersManager.java]: At this point yes. As it sounds a fatal error (i.e. corrupted file or similar).
Another thing is if a handler has some runtime error which is handled in the start() and error in logged but no hard stop.
(Good thing of this approach is this logic can be revisit if we thing it should be different).

**Comment by lucasponce** [comment]
@stefannegrea comments addressed, thanks for the review.

**Comment by lucasponce** [comment]
pom.xml fixed

**Comment by jshaughn** [comment]
I think the comments are handled, I'll merge so that @lucasponce can set a srcdep in his pending PR for alerts.  If necessary, further changes can be handled as add-ons.

**Pull request merged**